src/overlays/actors/ovl_En_Dnh/z_en_dnh.c

```diff
@@ -31,28 +55,130 @@ ActorInit En_Dnh_InitVars = {
+    } else {
+        play->nextEntrance = ENTRANCE(SOUTHERN_SWAMP_POISONED, 6);
+    }
+    gSaveContext.nextCutsceneIndex = 0;
+    play->transitionTrigger = TRANS_TRIGGER_START;
+    play->transitionType = TRANS_TYPE_FADE_WHITE;
+    gSaveContext.nextTransitionType = TRANS_TYPE_FADE_WHITE_SLOW;
+    CLEAR_WEEKEVENTREG(WEEKEVENTREG_90_40);
+    CLEAR_EVENTINF(EVENTINF_50);
+    return 1;
+}
+
+void* func_80A50DF8(EnDnh* this, PlayState* play) {
+    switch (this->unk198) {
+        case 1:
+            return D_80A51384;
+
+        case 2:
+            return D_80A5138C;
+
+        default:
+            return D_80A51250;
+    }
+}
+
+s32 func_80A50E40(EnDnh* this, PlayState* play) {
+    if (!(this->unk18C & 7) || !Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        return 0;
+    }
+    SubS_UpdateFlags(&this->unk18C, 0, 7);
+    this->msgEventScript = func_80A50DF8(this, play);
+    this->actionFunc = func_80A50F38;
+    return 1;
+}
+
+s32 func_80A50EC0(EnDnh* this) {
+    if (DECR(this->blinkTimer) == 0) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex >= 4) {
+            this->blinkTimer = Rand_S16Offset(30, 30);
+            this->eyeTexIndex = 0;
+        }
+    }
+}
+
+void func_80A50F38(EnDnh* this, PlayState* play) {
+    if (func_8010BF58(&this->actor, play, this->msgEventScript, this->msgEventCallback, &this->unk194)) {
+        SubS_UpdateFlags(&this->unk18C, 3, 7);
+        this->unk194 = 0;
+        this->unk198 = 0;
+        this->actionFunc = EnDnh_DoNothing;
+    }
+}
+
+void EnDnh_DoNothing(EnDnh* this, PlayState* play) {
+}
+
+void EnDnh_Init(Actor* thisx, PlayState* play) {
+    EnDnh* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_Init(play, &this->skelAnime, &object_tro_Skel_002950, NULL, this->jointTable, this->morphTable, 2);
+    SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, 0);
+    this->actor.shape.yOffset = 1100.0f;
+    if (gSaveContext.save.entrance != ENTRANCE(TOURIST_INFORMATION, 1)) {
+        SubS_UpdateFlags(&this->unk18C, 3, 7);
+        this->unk198 = 0;
+    } else {
+        SubS_UpdateFlags(&this->unk18C, 4, 7);
+        this->unk198 = CHECK_EVENTINF(EVENTINF_35) ? 2 : 1;
+    }
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_12_08)) {
+        this->actor.draw = NULL;
+    }
+    this->msgEventCallback = func_80A50D40;
+    this->unk194 = 0;
+    this->actionFunc = EnDnh_DoNothing;
+}
+
+void EnDnh_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void EnDnh_Update(Actor* thisx, PlayState* play) {
+    EnDnh* this = THIS;
+
+    func_80A50E40(this, play);
+    this->actionFunc(this, play);
+    func_80A50EC0(this);
+    SkelAnime_Update(&this->skelAnime);
+    func_8013C964(&this->actor, play, 60.0f, 30.0f, 0, this->unk18C & 7);
+    Actor_SetFocus(&this->actor, 26.0f);
+}
+
+s32 EnDnh_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    if (limbIndex == 1) {
```

Comment:
```suggestion
    if (limbIndex == OBJECT_TRO_LIMB_1) {
```
After you update the xml and re-extract.